Tests/InstrumentationTests/sdk-resource-extension-tests/ApplicationResourceProviderTests.swift

```diff
@@ -0,0 +1,80 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+import Foundation
+import OpenTelemetrySdk
+import XCTest
+
+@testable import ResourceExtension
+
+#if canImport(UIKit)
+    class ApplicationResourceProviderTests: XCTestCase {
+        func testContents() {
+            let appData = mockApplicationData(name: "appName", identifier: "com.bundle.id", version: "1.2.3", build: "9876")
+
+            let provider = ApplicationResourceProvider(source: appData) as ResourceProvider
+
+            let resource = provider.create()
+
+            XCTAssertTrue(appData.name == resource.attributes[ResourceAttributes.serviceName.rawValue]?.description)
```

Comment:
nit: should we use `XCTAssertEqual` in order to get better failure messages?

question: Are there any OTel guidelines with regards to the service name? I know internally we have a fairly strict format (so we would personally need a route to override this) but for example we don't support spaces or 'special characters' (basically non [a-z\-_]) in them. I'm not sure if there is anything in the spec for this.

Reply:
About the question you make, I think that it supports any AttributeValue, so the supported characters in the strings are not limited